### PR TITLE
EN-6373 : add test for ComputeEndOfEpochEconomics

### PR DESCRIPTION
**Pull request by miiu96**
- Add a test that verify if end of epoch economics data) is calculated as expected.

### CODE REVIEW

**Comment by elrondbigscam** [comment]
ERD BIG DUMP SELL NOW ...ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.

**Comment by elrondbigscam** [comment]
ERD BIG DUMP SELL NOW ...ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.

File: epochStart/metachain/economics_test.go [hunk 1]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
+		},
+		LeaderPercentageCalled: func() float64 {
+			return 0.1
+		},
+	}
+	args.RoundTime = &mock.RoundTimeDurationHandler{
+		TimeDurationCalled: func() time.Duration {
+			return time.Duration(roundDur) * time.Second
+		},
+	}
```
Comment (sasurobert) [inline epochStart/metachain/economics_test.go:473]: may create a function of setup.

**Comment by elrondscammer3** [comment]
Binance Delist/ Don't Buy ERD/ELROND / Erd scam coin / Erd big dump
ERD BIG DUMP SELL NOW ...ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.

**Comment by elrondscammer3** [comment]
ERD BIG DUMP SELL NOW ...ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.

**Comment by elrondscammer3** [comment]
..../ 
 Crooked sons of bitches. Give us our money back.
 BTC dump ERD dump BTC pump ERD dump BTC stable ERD dump. Don'T buy ER

File: epochStart/metachain/economics_test.go (hunk 1 above)
Comment (elronscam8) [reply epochStart/metachain/economics_test.go:473]: BTC dump ERD dump BTC pump ERD dump BTC stable ERD dump. Don'T buy ERD
  We want our money back, Romanian sons of bitches.
  
  Elrond was HACKED!
  Elrond announced that he had been hacked. Please don't buy ERD

**Comment by elronscam8** [comment]
That's enough!! Cheating bastards.
The price keeps falling. People don't trust you. ERD doesn't buy. What do you boys do for a living?

Binance Delist/ Don't Buy ERD/ELROND / Erd scam coin / Erd big dump
ERD BIG DUMP SELL NOW ...ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.
 
 Beniamin Mincu 
ERD/Elrond Scam Coin. Team ERD is a con man. Don't buy ERD. I send this
 message every day to everyone in the group. Prevent more people from being conned by sharing with you.
  ERD BIG DUMP SELL NOW ...
 
 
 Crooked sons of bitches. Give us our money back.
 BTC dump ERD dump BTC pump ERD dump BTC stable ERD dump. Don'T buy ERD
 
  BTC dump ERD dump BTC pump ERD dump BTC stable ERD dump. Don'T buy ERD
  We want our money back, Romanian sons of bitches.
  
  Elrond was HACKED!
  Elrond announced that he had been hacked. Please don't buy ERD

File: epochStart/metachain/economics_test.go [hunk 2]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
+
+	epochDuration := numberOfSecondsInDay
+	roundsPerEpoch := uint64(epochDuration / roundDur)
+	type testInput struct {
+		blockPerEpochOneShard uint64
+		acFeesInEpoch         *big.Int
+		devFeesInEpoch        *big.Int
+	}
+
+	testInpus := []testInput{
```
Comment (SebastianMarian) [inline epochStart/metachain/economics_test.go:513]: testInputs

File: epochStart/metachain/economics_test.go [hunk 3]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
 }
 
+func TestComputeEndOfEpochEconomics(t *testing.T) {
+	t.Parallel()
+
+	commAddress := "communityAddress"
+	totalSupply, _ := big.NewInt(0).SetString("20000000000000000000000000000", 10) // 20 Billions ERD
+	nodePrice, _ := big.NewInt(0).SetString("1000000000000000000000", 10)          // 1000 ERD
+
+	roundDur := 4
```
Comment (SebastianMarian) [inline epochStart/metachain/economics_test.go:453]: roundDuration

File: epochStart/metachain/economics_test.go [hunk 4]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
+		},
+	}
+
+	ec, _ := NewEndOfEpochEconomicsDataCreator(args)
+
+	epochDuration := numberOfSecondsInDay
+	roundsPerEpoch := uint64(epochDuration / roundDur)
+	type testInput struct {
+		blockPerEpochOneShard uint64
+		acFeesInEpoch         *big.Int
```
Comment (SebastianMarian) [inline epochStart/metachain/economics_test.go:509]: What is ac ? actual? rename it to be better understood

File: epochStart/metachain/economics_test.go [hunk 5]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
+					{ShardID: 0, Round: roundsPerEpoch, Nonce: input.blockPerEpochOneShard},
+					{ShardID: 1, Round: roundsPerEpoch, Nonce: input.blockPerEpochOneShard},
+				},
+			},
+		}
+
+		economicsBlock, err := ec.ComputeEndOfEpochEconomics(meta)
+		assert.Nil(t, err)
+
+		blocksPerEpochTotal := int64(input.blockPerEpochOneShard * 3)
```
Comment (SebastianMarian) [inline epochStart/metachain/economics_test.go:543]: totalBlocksPerEpoch

File: epochStart/metachain/economics_test.go (hunk 4 above)
Comment (miiu96) [reply epochStart/metachain/economics_test.go:509]: renamed.

File: epochStart/metachain/economics_test.go (hunk 3 above)
Comment (miiu96) [reply epochStart/metachain/economics_test.go:453]: renamed.

File: epochStart/metachain/economics_test.go (hunk 2 above)
Comment (miiu96) [reply epochStart/metachain/economics_test.go:513]: done.

File: epochStart/metachain/economics_test.go (hunk 5 above)
Comment (miiu96) [reply epochStart/metachain/economics_test.go:543]: done.

File: epochStart/metachain/economics_test.go [hunk 6]
```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
...
+		{blockPerEpochOneShard: roundsPerEpoch / 4, acFeesInEpoch: intToErd(1000), devFeesInEpoch: intToErd(1000)},
+		{blockPerEpochOneShard: roundsPerEpoch / 8, acFeesInEpoch: intToErd(1000), devFeesInEpoch: intToErd(1000)},
+		{blockPerEpochOneShard: roundsPerEpoch / 16, acFeesInEpoch: intToErd(1000), devFeesInEpoch: intToErd(1000)},
+		{blockPerEpochOneShard: roundsPerEpoch / 32, acFeesInEpoch: intToErd(1000), devFeesInEpoch: intToErd(1000)},
+		{blockPerEpochOneShard: roundsPerEpoch / 64, acFeesInEpoch: intToErd(100000000000000), devFeesInEpoch: intToErd(10000000)},
+		{blockPerEpochOneShard: roundsPerEpoch, acFeesInEpoch: intToErd(100000000000000), devFeesInEpoch: intToErd(30000000000000)},
+	}
+
+	rewardsPerBlock, _ := big.NewInt(0).SetString("84559445290038908043", 10) // *based on 0.1 inflation
+	for _, input := range testInpus {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply epochStart/metachain/economics_test.go:525]: done.

File: epochStart/metachain/economics_test.go (hunk 1 above)
Comment (miiu96) [reply epochStart/metachain/economics_test.go:473]: done.

**Pull request merged**